staging/src/k8s.io/legacy-cloud-providers/vsphere/nodemanager.go

```diff
@@ -201,7 +204,21 @@ func (nm *NodeManager) DiscoverNode(node *v1.Node) error {
 				if vm != nil {
 					klog.V(4).Infof("Found node %s as vm=%+v in vc=%s and datacenter=%s",
 						node.Name, vm, res.vc, res.datacenter.Name())
-
+					var vmconfig *types.VirtualMachineConfigInfo
+					err := vm.Properties(ctx, vm.Reference(), []string{"config"}, &vmconfig)
+					if err != nil || vmconfig == nil {
```

Comment:
Will this result in any API calls btw? Will this check happen from control-plane too?

Reply:
> Will this result in any API calls btw?

yes, this will call property collector -> RetrieveProperties API on the vCenter.

> Will this check happen from control-plane too?

Can you help explain this question?

Reply:
I was just wondering if this will result in N new API calls to vcenter for each node in the cluster.

Reply:
Yes during node discovery time only, we will make these calls to find the Guest VM Hardware version.  We can not get this detail without an additional VC API call.